### PR TITLE
feat: support for init and storing of dataset subclasses

### LINKED ISSUE #7: DatasetRegistry
Create a static class containing all dataset implementations.

Requires #5, #6

### PR DESCRIPTION
**Pull request by tediiieee**
Closes #7

### CODE REVIEW

File: src/dataset/DatasetRegistry.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    _DATASETS = None
+
+    @staticmethod
+    def init_registry():
+        """
+        Static method to initialize `DATASETS` with Dataset subclasses.
+        Calls the constructor for each dataset with predefined parameters.
+
+        If more dataset are implemented they should be places in `DATASETS`.
```
Comment (sporreking) [inline src/dataset/DatasetRegistry.py:19]: "...they should be added within this function." maybe

File: src/dataset/DatasetRegistry.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        """
+        DatasetRegistry._DATASETS = dict(FFHQ_256=FFHQDataset(256))
+
+    @staticmethod
+    def get_names() -> list:
```
Comment (sporreking) [inline src/dataset/DatasetRegistry.py:24]: `-> list[str]`

same goes for docstring

File: src/dataset/DatasetRegistry.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        """
+        Returns a dataset with the given `name` from the registry.
+
+        Args:
+            name (str): Name of the dataset.
+
+        Returns:
+            Dataset: Dataset with the given `name`.
+        """
+        return list(DatasetRegistry._DATASETS[name])
```
Comment (sporreking) [inline src/dataset/DatasetRegistry.py:44]: why list?

File: src/dataset/DatasetRegistry.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    @staticmethod
+    def get_datasets() -> list:
```
Comment (sporreking) [inline src/dataset/DatasetRegistry.py:47]: `list[Dataset]`

also in docstring

**Review by sporreking — CHANGES_REQUESTED** [review]
Looks good overall. There are a few comments though.

Also, the commit message works but could be condensed more, i.e., "feat: dataset storage support".

File: src/dataset/DatasetRegistry.py (hunk 3 above)
Comment (tediiieee) [reply src/dataset/DatasetRegistry.py:44]: should not be list

**Review by sporreking — APPROVED** [review]
Nice work!

**Pull request merged**